sdks/java/extensions/sketching/src/test/java/org/apache/beam/sdk/extensions/sketching/TDigestQuantilesTest.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.Values;
+import org.apache.beam.sdk.transforms.WithKeys;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/** Tests for {@link TDigestQuantiles}. */
+public class TDigestQuantilesTest {
+
+  @Rule public final transient TestPipeline tp = TestPipeline.create();
+
+  private static final List<Double> stream = generateStream();
+
+  private static final int size = 999;
+
+  private static final int compression = 100;
+
+  private static List<Double> generateStream() {
+    List<Double> li = new ArrayList<>();
+    for (double i = 1D; i <= size; i++) {
+      li.add(i);
+    }
+    Collections.shuffle(li);
+    return li;
+  }
+
+  @Test
+  public void globally() {
+    PCollection<KV<Double, Double>> col = tp.apply(Create.of(stream))
+            .apply(TDigestQuantiles.globally().withCompression(compression))
+            .apply(ParDo.of(new RetrieveQuantiles(0.25, 0.5, 0.75, 0.99)));
+
+    PAssert.that("Verify Accuracy", col).satisfies(new VerifyAccuracy());
+    tp.run();
+  }
+
+  @Test
+  public void perKey() {
+    PCollection<KV<Double, Double>> col = tp.apply(Create.of(stream))
+            .apply(WithKeys.<Integer, Double>of(1))
+            .apply(TDigestQuantiles.<Integer>perKey().withCompression(compression))
+            .apply(Values.<MergingDigest>create())
+            .apply(ParDo.of(new RetrieveQuantiles(0.25,  0.5, 0.75, 0.99)));
+
+    PAssert.that("Verify Accuracy", col).satisfies(new VerifyAccuracy());
+
+    tp.run();
+  }
+
+  @Test
+  public void testCoder() throws Exception {
+    MergingDigest tDigest = new MergingDigest(1000);
+    for (int i = 0; i < 10; i++) {
+      tDigest.add(2.4 + i);
+    }
+
+    Assert.assertTrue("Encode and Decode", encodeDecodeEquals(tDigest));
+  }
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testMergeAccum() {
+      Random rd = new Random(1234);
+      List<MergingDigest> accums = new ArrayList<>();
+      for (int i = 0; i < 3; i++) {
+          MergingDigest std = new MergingDigest(100);
+          for (int j = 0; j < 1000; j++) {
+              std.add(rd.nextDouble());
+          }
+          accums.add(std);
+      }
+      TDigestQuantilesFn fn = TDigestQuantilesFn.create(100);
+      MergingDigest res = fn.mergeAccumulators(accums);
+  }
+
+  private <T> boolean encodeDecodeEquals(MergingDigest tDigest) throws IOException {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
Can you use `CoderProperties` instead?

Reply:
No, T-Digest seems to not be consistent with equals (I tried)

Reply:
Oh ok. I recommend at least using `CoderUtils.clone()` instead of lines 121-128 then.

Reply:
Done, thanks.

Reply:
Doesn't seem done, did you push the change?